### PR TITLE
[NES-212] make YieldToken implement IComponentToken

### PR DESCRIPTION
**Pull request by eyqs**
## What's new in this PR?

This allows us to put YieldTokens into AggregateTokens.

## Why?

What problem does this solve?
Why is this important?
What's the context?

### CODE REVIEW

File: smart-wallets/src/token/YieldToken.sol [hunk 1]
```diff
@@ -96,6 +170,66 @@ contract YieldToken is YieldDistributionToken, WalletUtils, IYieldToken {
         _mint(user, yieldTokenAmount);
```
Comment (ungaro) [inline smart-wallets/src/token/YieldToken.sol:170]: should we change this function name to `mintByOwner`

File: smart-wallets/src/token/YieldToken.sol [hunk 2]
```diff
@@ -96,6 +170,66 @@ contract YieldToken is YieldDistributionToken, WalletUtils, IYieldToken {
...
+        uint256 assets
+    ) public view override(ERC4626, IComponentToken) returns (uint256 shares) {
+        return assets;
+    }
+
+    /// @inheritdoc IERC4626
+    function convertToAssets(
+        uint256 shares
+    ) public view override(ERC4626, IComponentToken) returns (uint256 assets) {
+        return shares;
```
Comment (ungaro) [inline smart-wallets/src/token/YieldToken.sol]:
```
    uint256 supply = totalSupply();
    
    if (supply == 0) {
        return shares;
    }
    
    return (shares * totalAssets()) / supply;
```

File: smart-wallets/src/token/YieldToken.sol [hunk 3]
```diff
@@ -96,6 +170,66 @@ contract YieldToken is YieldDistributionToken, WalletUtils, IYieldToken {
...
+    /// @inheritdoc IERC4626
+    function totalAssets() public view override(ERC4626, IComponentToken) returns (uint256 totalManagedAssets) {
+        return super.totalAssets();
+    }
+
+    /// @inheritdoc IERC4626
+    function convertToShares(
+        uint256 assets
+    ) public view override(ERC4626, IComponentToken) returns (uint256 shares) {
+        return assets;
```
Comment (ungaro) [inline smart-wallets/src/token/YieldToken.sol]:
```
    uint256 supply = totalSupply();
    uint256 totalAssets_ = totalAssets();
    
    if (supply == 0 || totalAssets_ == 0) {
        return assets;
    }
    
    return (assets * supply) / totalAssets_;
```

**Review by ungaro — APPROVED** [review]
there are some outstanding comments to be reviewed.

File: smart-wallets/src/token/YieldToken.sol (hunk 1 above)
Comment (eyqs) [reply smart-wallets/src/token/YieldToken.sol:170]: renamed to `adminMint`

**Pull request merged**